src/ape/managers/chain.py

```diff
@@ -0,0 +1,360 @@
+        Only providers that implement `EIP-1559 <https://eips.ethereum.org/EIPS/eip-1559>`__
+        will use this property.
+
+        Raises:
+            NotImplementedError: When this provider does not implement
+              `EIP-1559 <https://eips.ethereum.org/EIPS/eip-1559>`__.
+        """
+
+        return self.provider.base_fee
+
+    @property
+    def pending_timestamp(self) -> int:
+        """
+        The current epoch time of the chain, as an ``int``.
+        """
+
+        return int(time.time() + self._time_offset)
+
+    def snapshot(self) -> SnapshotID:
+        """
+        Record the current state of the blockchain with intent to later
+        call the method :meth:`~ape.managers.chain.ChainManager.revert`
+        to go back to this point. This method is for development networks
+        only.
+
+        Raises:
+            NotImplementedError: When the active provider does not support
+              snapshotting.
+
+        Returns:
+            :class:`~ape.types.SnapshotID`: The snapshot ID.
+        """
+        provider = self._get_test_provider(TestProviderAPI.snapshot.__name__)
+        snapshot_id = provider.snapshot()
+
+        if snapshot_id not in self._snapshots:
+            self._snapshots.append(snapshot_id)
+
+        return snapshot_id
+
+    def restore(self, snapshot_id: Optional[SnapshotID] = None):
+        """
+        Regress the current call using the given snapshot ID.
+        Allows developers to go back to a previous state.
+
+        Raises:
+            NotImplementedError: When the active provider does not support
+              snapshotting.
+            :class:`~ape.exceptions.UnknownSnapshotError`: When the snapshot ID is not cached.
+            :class:`~ape.exceptions.ChainError`: When there are no snapshot IDs to select from.
+
+        Args:
+            snapshot_id (Optional[:class:`~ape.types.SnapshotID`]): The snapshot ID. Defaults
+              to the most recent snapshot ID.
+        """
+        provider = self._get_test_provider(TestProviderAPI.revert.__name__)
+        if not self._snapshots:
+            raise ChainError("There are no snapshots to revert to.")
+        elif snapshot_id is None:
+            snapshot_id = self._snapshots.pop()
+        elif snapshot_id not in self._snapshots:
+            raise UnknownSnapshotError(snapshot_id)
+        else:
+            snapshot_index = self._snapshots.index(snapshot_id)
+            self._snapshots = self._snapshots[:snapshot_index]
+
+        provider.revert(snapshot_id)
+
+    def _get_test_provider(self, method_name: str) -> TestProviderAPI:
+        provider = self.provider
+        if not isinstance(provider, TestProviderAPI):
+            raise NotImplementedError(
+                f"Provider '{provider.name}' does not support method '{method_name}'."
+            )
+
+        return provider
+
+
+def _convert(*args, **kwargs):
+    from ape import convert
+
+    return convert(*args, **kwargs)
```

Comment:
Can this be a `cached_property` on `AccountHistory`?

Reply:
Yes!

But, I am pretty sure lazy imports are automatically cached. At least it seems like it

Reply:
Oh yes, that is true. `importlib` caches imports